### PR TITLE
React grader sum chart

### PR DESCRIPTION
**Pull request by devan-srinivasan**
## Motivation and Context
This PR updated the existing 'Grades Summary' chart to use `react chartjs-2`

## Your Changes

**Description**:
I changed the ajax request to use `fetch` instead so as to match the other components. [More to be added shortly]

**Type of change** (select all that apply):

 

- [x] Refactoring (internal change to codebase, without changing functionality)

## Testing

I ran the `rspec` test suite and tested my chart in the UI. Specifically the following 3 usages:
1) Grades Summary Tab
2) Marks Chart from the submission of a group under the 'Summary' tab
3) The grades for an assignment from a student's client

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have fixed any Hound bot comments.

### CODE REVIEW

**Comment by devan-srinivasan** [comment]
Hope this is okay. Let me know if anything needs to be changed...

File: app/assets/javascripts/Components/Helpers/data_chart.jsx [hunk 1]
```diff
@@ -45,30 +39,8 @@ export class DataChart extends React.Component {
...
-    this.chart.options.legend.fontColor = document.documentElement.style.getPropertyValue('--line');
-    this.chart.options.scales.y.scaleLabel = {display: true, labelString: this.props.yTitle};
-    this.chart.options.scales.x.scaleLabel = {display: true, labelString: this.props.xTitle};
-    this.chart.update();
-  }
-
-  render() {
     return (
-      <canvas id='term_marks' style={{display: 'inline-flex', width: this.props.width, height: 500, margin: '10px'}}></canvas>
+      <Bar data={this.props} options={options} width={1000} height={450}/>
```
Comment (david-yz-liu) [inline app/assets/javascripts/Components/Helpers/data_chart.jsx:43]: You have to be careful when changing components that could be used in multiple places, and test *all* places where the component is used.

In particular, you removed some existing styling, and also removed the use of `this.props.width`, which is important to allow customization.

File: app/assets/javascripts/Components/Helpers/data_chart.jsx [hunk 2]
```diff
@@ -1,16 +1,10 @@
 import React from 'react';
 import {render} from 'react-dom';
+import { Bar } from 'react-chartjs-2'
```
Comment (david-yz-liu) [inline app/assets/javascripts/Components/Helpers/data_chart.jsx:3]: Add a semicolon here.

File: app/assets/javascripts/Components/Helpers/data_chart.jsx (hunk 1 above)
Comment (devan-srinivasan) [reply app/assets/javascripts/Components/Helpers/data_chart.jsx:43]: Oops okay. Will fix.

**Comment by devan-srinivasan** [comment]
I implemented your comments. With auto the width is kind of weird let me know what you think. I find it too narrow while using `auto`

**Comment by david-yz-liu** [comment]
Thank you, @devan-srinivasan! I agreed that the `'auto'` made the chart too narrow, so I played around with the settings to get the graph to be responsive and automatically fill its container.

**Comment by devan-srinivasan** [comment]
@david-yz-liu Awesome!

**Pull request merged**